src/common/transformations/include/transformations/op_conversions/reduce_modification_for_false_keepdims.hpp

```diff
@@ -0,0 +1,155 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <algorithm>
+#include <memory>
+#include <openvino/core/rt_info.hpp>
+#include <openvino/opsets/opset1.hpp>
+#include <openvino/pass/graph_rewrite.hpp>
+#include <transformations_visibility.hpp>
+#include <vector>
+
+namespace ov {
+namespace pass {
+
+// Modify reduce if keep_dims is false and it reduces batch and spatial axes
+class TRANSFORMATIONS_API ModifyReduceForFalseKeepDims;
+class TRANSFORMATIONS_API ModifyReduceMeanForFalseKeepDims;
```

Comment:
I believe we need only one MatcherPass:
```
pattern::wrap_type<opset10::ReduceSum,  opset10::ReduceMean,  opset10::ReduceLogicalAnd, 
                                opset10::ReduceLogicalOr,  opset10::ReduceMin, opset10::ReduceMax> 
                                 ({pattern::any_input(pattern::has_static_shape()), pattern::wrap_type<opset1::Constant>()},  
                                  pattern::has_static_shape()),  matcher_name);
```
in `modify_reduce_for_false_keepdims` we can cast `m.get_match_root()` to `ArithmeticReductionKeepDims` amd `LogicalReductionKeepDims` base classes to set `keep_dims` flag

and we can replace the old reduce_const with new one instead of creating new Reduce operation by
`node->input(1).replace_source_output(new_reduce_const);`

After that we won't need `template` and we will be able to move `modify_reduce_for_false_keepdims` to .cpp file

Reply:
Applied